OSDOCS-5637-bugbatch1: Documented batch1 4.13 bug fix release notes (#59011)
openshift/openshift-docs: release_notes/ocp-4-13-release-notes.adoc

For more information see xref:../scalability_and_performance/cnf-talm-for-cluster-upgrades.adoc#talo-precache-feature-image-filter_cnf-topology-aware-lifecycle-manager[Using the container image pre-cache filter].

// Jira for this enhancement = https://issues.redhat.com/browse/OCPBUGS-3098
[id="ocp-4-13-insights-operator"]

> === Updating to etcd server version 3.5.7

Author: CC: @tmalove

> === Updating to etcd server version 3.5.7

Author: Thanks, all. I'll drop this note and updated the Jira accordingly.

[id="ocp-4-13-insights-operator"]
=== Insights Operator
// https://issues.redhat.com/browse/OCPBUGS-6832

> The Insights Operator can now collect the `openshift_apps_deploymentconfigs_strategy_total` metric. This metric gathers deployment strategy information from a deployment's configuration.

Reviewer: ```suggestion
With this release, the Insights Operator collects the `openshift_apps_deploymentconfigs_strategy_total` metric. This metric gathers deployment strategy information from a deployment's configuration.
```

Author: I'm unsure of this as the "release" is already implied.

[id="ocp-4-13-bare-metal-hardware-bug-fixes"]
==== Bare Metal Hardware Provisioning

> * Previously, when you attempted to deploy an {product-title} cluster node on a server that is configured with the Integrated Lights-Out (iLO) Management Interface Driver, provisioning of the node would fail. You would receive an error message similar to "An auth plugin is required to determine endpoint URL". This issue occurs because of a missing `[ilo]/use_web_server_for_images` configuration parameter in the iLO driver that causes the driver to attempt to use object storage as the default storage mechanism. Object storage is not present in the product. The {product-title} 4.13 version and later versions fix this issue by including `[ilo]/use_web_server_for_images` in the iLO driver's configuration, which ensures that the iLO driver uses a web server running in the metal3 pod. (link:https://issues.redhat.com/browse/OCPBUGS-5068[*OCPBUGS-5068*])

Author: "Deploying" would not work because the dependant clause also begins with a gerund, "provisioning".

Guidance used to exist about using "because" over "due", so I used "because" to be safe.

I used "present" in the verb form so I retained the "in the product" clause. 

Removed "fixing the failire2 as it dangles a bit.

------------------
All other changes implemented.

* Previously, if you configured the rendezvous IP to the IP address of the host that has a `role` parameter set to `worker` and you generated an ISO image, the Agent-based installer would fail to install the cluster. Now, when you attempt to generate an ISO image based on this configuration, you will receive a validation failure message. On receiving this message, you must update the `rendezvousIP` field in the `agent-config.yaml` file to use the IP of a host with the `master` role. (link:https://issues.redhat.com/browse/OCPBUGS-2088[*OCPBUGS-2088*])

> * Previously, the `openshift-installer ` program did not accept the following new regions defined in the `aws-sdk-go` library: `ap-south-2`, `ap-southeast-4`, `eu-central-2`, `eu-south-2`, and `me-central-1`. When you used the program to create the installation configuration file, the program would not list these new regions or accept any manually entry attempts for these regions. Now, the `openshift-installer ` program supports these regions and you can specify them when you create the installation configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-10213[*OCPBUGS-10213*])

Author: Is "Now" incorrect to use? I'm not seeing any guidance that discourages its use? 
---------------
Updated the rest of the text.

> * Previously, the `openshift-installer ` program did not accept the following new regions defined in the `aws-sdk-go` library: `ap-south-2`, `ap-southeast-4`, `eu-central-2`, `eu-south-2`, and `me-central-1`. When you used the program to create the installation configuration file, the program would not list these new regions or accept any manually entry attempts for these regions. Now, the `openshift-installer ` program supports these regions and you can specify them when you create the installation configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-10213[*OCPBUGS-10213*])

Author: Example in the Documentation guidelines GH page:

![Screenshot from 2023-04-28 11-20-49](https://user-images.githubusercontent.com/57954076/235122381-a137228c-0137-4e6d-a852-c3b79ae0b4bf.png)

> * Previously, the `openshift-installer ` program did not accept the following new regions defined in the `aws-sdk-go` library: `ap-south-2`, `ap-southeast-4`, `eu-central-2`, `eu-south-2`, and `me-central-1`. When you used the program to create the installation configuration file, the program would not list these new regions or accept any manually entry attempts for these regions. Now, the `openshift-installer ` program supports these regions and you can specify them when you create the installation configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-10213[*OCPBUGS-10213*])

Reviewer: > Is "Now" incorrect to use? I'm not seeing any guidance that discourages its use?
> 
> Updated the rest of the text.

No, I don't think it is incorrect. I just suggested "with this update" because that keeps consistency with other bug's text.

> * Previously, an issue exists with the code base that sets `Machine.PrimaryNetwork` based on the`controlPlane.platform.openstack.failureDomain` field in the `install-config.yaml` file. Specifically, when you set a `control-plane` port in a `failureDomain` Technology Preview component in your installation configuration file, the installation program continues to use a default network subnet, or `platform.openstack.machinesSubnet` if present, when the program should use `controlPlane.platform.openstack.failureDomain[].portTarget[id==control-plane]`. This issue impacts {product-title} that runs with Kuryr from identifying the port on a {rh-openstack-first} subnet that control plane machines use for communicating between them. Now, when you set `control-plane` for `portTarget` in the `failureDomain` Technology Preview component, the installation program sets the port's information in the `Machine.PrimaryNetwork` field, so that your {product-title} cluster can succesfully run with Kuryr. (link:https://issues.redhat.com/browse/OCPCLOUD-10658[*OCPCLOUD-10658*])

Reviewer: This was a lot of text and I'm not sure the user needed to the level of specification in that one sentence. I cut for minimalism.

> * Previously, an issue exists with the code base that sets `Machine.PrimaryNetwork` based on the`controlPlane.platform.openstack.failureDomain` field in the `install-config.yaml` file. Specifically, when you set a `control-plane` port in a `failureDomain` Technology Preview component in your installation configuration file, the installation program continues to use a default network subnet, or `platform.openstack.machinesSubnet` if present, when the program should use `controlPlane.platform.openstack.failureDomain[].portTarget[id==control-plane]`. This issue impacts {product-title} that runs with Kuryr from identifying the port on a {rh-openstack-first} subnet that control plane machines use for communicating between them. Now, when you set `control-plane` for `portTarget` in the `failureDomain` Technology Preview component, the installation program sets the port's information in the `Machine.PrimaryNetwork` field, so that your {product-title} cluster can succesfully run with Kuryr. (link:https://issues.redhat.com/browse/OCPCLOUD-10658[*OCPCLOUD-10658*])

Author: Good point. I like your suggested text. I changed "exists" to "existed" in the first sentence to maintain a setnence with past tense.

* Previously, if you configured the rendezvous IP to the IP address of the host that has a `role` parameter set to `worker` and you generated an ISO image, the Agent-based installer would fail to install the cluster. Now, when you attempt to generate an ISO image based on this configuration, you will receive a validation failure message. On receiving this message, you must update the `rendezvousIP` field in the `agent-config.yaml` file to use the IP of a host with the `master` role. (link:https://issues.redhat.com/browse/OCPBUGS-2088[*OCPBUGS-2088*])

* Previously, the `openshift-installer` program did not accept the following new regions defined in the `aws-sdk-go` library: `ap-south-2`, `ap-southeast-4`, `eu-central-2`, `eu-south-2`, and `me-central-1`. When you used the installation program to create the installation configuration file, the installation program would not list these new regions or accept manual entries for these regions. With this update, the `openshift-installer ` program supports these regions and you can specify them when you create the installation configuration file. (link:https://issues.redhat.com/browse/OCPBUGS-10213[*OCPBUGS-10213*])

> * Previously, an issue existed with the code base that sets `Machine.PrimaryNetwork` based on the`controlPlane.platform.openstack.failureDomain` field in the `install-config.yaml` file. This issue impacts {product-title} that runs with Kuryr from identifying the port on a {rh-openstack-first} subnet that control plane machines use for communicating between them. With this update, when you set `control-plane` for `portTarget` in the `failureDomain` Technology Preview component, the installation program sets the port's information in the `Machine.PrimaryNetwork` field, so that your {product-title} cluster successfully runs with Kuryr.(link:https://issues.redhat.com/browse/OCPCLOUD-10658[*OCPCLOUD-10658*])

Reviewer: The link doesn't work. Should this be OCPBUGS?

[id="ocp-4-13-storage-bug-fixes"]
==== Storage

> * Previously, the `openshift-manila-csi-driver` namespace did not include labels that are required for the management of workload partitioning. These missing labels can impact the operation of restricting Manila CSI pods to run on a selected set of CPUs. The {product-title} {product-version} fixes this issue in that the `openshift-manila-csi-driver` namespace now includes the `workload.openshift.io/allowed` label. (link:https://issues.redhat.com/browse/OCPBUGS-11341[*OCPBUGS-11341*])

Reviewer: ```suggestion
* Previously, the `openshift-manila-csi-driver` namespace did not include labels that are required for the management of workload partitioning. These missing labels impacted the operation of restricting Manila CSI pods to run on a selected set of CPUs. With this update, the `openshift-manila-csi-driver` namespace now includes the `workload.openshift.io/allowed` label. (link:https://issues.redhat.com/browse/OCPBUGS-11341[*OCPBUGS-11341*])
```

* Currently, a certificate compliance issue, specifically outputted as `x509: certificate is not standards compliant`, exists when you run the `openshift-installer` on macOS for the purposes of installing an {product-title} cluster on VMware vSphere. This issue relates to a known issue with the `golang` compiler in that the compiler does not recognize newly supported macOS certificate standards. No workaround exists for this issue. (link:https://issues.redhat.com/browse/OSDOCS-5694[*OSDOCS-5694*])

> * The load balancing algorithm did not prioritize existing control plane machines when more than three failure domains were included in the `ControlPlaneMachineSet` definition. If you added a fourth failure domain that was alphabetically higher in precedence than the existing three failure domains to the definition, this fourth failure domain take precedence over any existing failure domains. This behavior would cause rolling forward updates to be applied to a control plane machine. To avoid this scenario, ensure that existing in-use failure domains retain a higher precedence than the new and unused failure domains. By completing this action, each control plane machine remains stable during the course of adding more than three failure domains to the definition. (link:https://issues.redhat.com/browse/OCPBUGS-11968[*OCPBUGS-11968*])

Reviewer: ```suggestion
* When more than three failure domains are included in the `ControlPlaneMachineSet` definition, the load balancing algorithm does not prioritize existing control plane machines. If you add a fourth failure domain that is alphabetically higher in precedence than the existing three failure domains to the definition, the fourth failure domain takes precedence over any existing failure domains. This behavior causes rolling forward updates to be applied to a control plane machine. As a workaround, set existing in-use failure domains to a higher precedence than the new and unused failure domains. This workaround stabilizes control plane machines during the course of adding more than three failure domains to the definition. (link:https://issues.redhat.com/browse/OCPBUGS-11968[*OCPBUGS-11968*])
```

Reviewer: Present tense and focused on actions the user can take.

Author: Removed past tense and passive voice. 

"Workaround" is classed as jargon, so I did not implement this change:
![Screenshot from 2023-04-28 10-36-57](https://user-images.githubusercontent.com/57954076/235112953-50859e96-636e-48dc-98dd-3d41b18b09ce.png)
